compose/service.py

```diff
@@ -333,7 +333,14 @@ def start_container(self, container, intermediate_container=None, **override_opt
             dns_search=dns_search,
             restart_policy=restart,
             cap_add=cap_add,
-            cap_drop=cap_drop,
+            cap_drop=cap_drop
+        )
+
+    def start_container(self, container, intermediate_container=None, **override_options):
+        container = container or self.create_container(**override_options)
```

Comment:
This line still needs removing. We should always expect `container` to not be None.